tests/cpp/merge_mps/merge_mps_test.cpp

```diff
@@ -151,63 +176,114 @@ BOUNDS
     UP BOUND      Y1        50.0
     UP BOUND      Y2        50.0
 ENDATA)";
-    slave1.close();
-    options_.weights["slave1.mps"] = 1;
+    satellite1.close();
+
+    //Required to avoid missing contribution to Obj
+    options_.weights["satellite1.mps"] = 1;
+
     std::filesystem::current_path(tmp_dir_);
     MergeMPS mergeMPS(options_, logger_, writer_);
     mergeMPS.launch();
-    //Print the merged problem
-    std::ifstream merged_mps(tmp_dir_ / "log_merged.mps"s);
-    std::string line;
-    while (std::getline(merged_mps, line)) {
-        std::cout << line << std::endl;
-    }
-    std::cout << "-------\n";
-    std::ifstream merged_lp(tmp_dir_ / "log_merged.lp"s);
-    std::string line2;
-    while (std::getline(merged_lp, line2)) {
-        std::cout << line2 << std::endl;
-    }
+
     auto lastSolution = writer_->solution_data_;
     EXPECT_EQ(lastSolution.problem_status, "OPTIMAL");
-    //log_merged_mps exists
+
     EXPECT_TRUE(std::filesystem::exists(tmp_dir_ / "log_merged.mps"s));
     EXPECT_TRUE(std::filesystem::exists(tmp_dir_ / "log_merged.lp"s));
+
     SolverFactory factory;
     auto merged = factory.create_solver("CBC");
-    merged->read_prob_mps(tmp_dir_ / "log_merged.lp"s);
+    merged->read_prob_mps(tmp_dir_ / "log_merged.mps"s);
+
     auto master = factory.create_solver("CBC");
     master->read_prob_mps(tmp_dir_ / "master.mps"s);
-    auto slave = factory.create_solver("CBC");
-    slave->read_prob_mps(tmp_dir_ / "slave1.mps"s);
-    //Merged has master + slave number of constraints
-    EXPECT_EQ(merged->get_nrows(), master->get_nrows() + slave->get_nrows());
-    //Merged has master + slave number of variables
-    EXPECT_EQ(merged->get_ncols(), master->get_ncols() + slave->get_ncols());
-    //Merged obj is the combination of master and slave
+
+    auto satellite = factory.create_solver("CBC");
+    satellite->read_prob_mps(tmp_dir_ / "satellite1.mps"s);
+
+    //Verify dimensions
+    EXPECT_EQ(merged->get_nrows(), master->get_nrows() + satellite->get_nrows());
```

Comment:
Just use the utility functions from `SolverAbstract`, you have worked hard to introduce ;)